iree/compiler/Codegen/SPIRV/KernelDispatchUtils.cpp

```diff
@@ -339,15 +337,24 @@ LogicalResult setDefaultRootConfig(FuncOp entryPoint,
   }
 
   std::array<int64_t, 3> workgroupSize = {subgroupSize, 1, 1};
+
   unsigned loopDepth = partitionedLoops.back() + 1;
-  SmallVector<int64_t, 4> workgroupTileSize(loopDepth, 1),
-      threadTileSize(loopDepth, 1);
+  SmallVector<int64_t, 4> workgroupTileSize(loopDepth, 0);
+  SmallVector<int64_t, 4> threadTileSize(loopDepth, 0);
+
+  // Tile along the last 3 parallel dimensions.
+  for (int i = 1; i < std::min(3u, loopDepth); ++i) {
```

Comment:
Dont want to hard code this. I have a utility method `getPartitionedLoops` which returns which loops are partitioned. Could we use that instead?